app/code/Magento/Ui/view/base/web/js/grid/columns/date.js

```diff
@@ -15,7 +15,7 @@ define([
 
     return Column.extend({
         defaults: {
-            dateFormat: 'MMM d, YYYY h:mm:ss A'
+            dateFormat: 'MMM d, YYYY h:mm:ss'
```

Comment:
@ArtiDjeims In the current case, we use h:mm:ss format, in this the case 'h' means that we use 12h time format, so we should use A to show PM/AM abbreviation. 
You can look documentation of monent.js library that we use for work with a date and time.
https://momentjs.com/docs/#the-rfc-2822-date-time-format